### ISSUE TITLE
Override/mock scope definitions

### ISSUE THREAD

**Issue by Yazazzello** [InsertKoinIO/koin#450]
**Is your feature request related to a problem? Please describe.**
for testing mostly
**Describe the solution you'd like**
Ability to mock or override scope definitions, preferably the same way as for regular definitions

**Describe alternatives you've considered**
I tried replace manually, with scope `beanregistry` -> `createMockedDefinition`, but couldn't save it with `saveDefinition` even with option override = true
**Target Koin project**
org.koin:koin-androidx-scope
2.1.0 ?

**Comment by arnaudgiuliani** [comment]
@Yazazzello did you tried with `declareMock` and `declare` from `koin-test` & `koin-core`?

**Comment by sadeqzadeh** [comment]
Hey @arnaudgiuliani! Is there any update on this? I saw the plan to fix this in 2.1.0, but this is a blocker for us right now as we have invested some time in integrating Koin and our tests are broken.

Our current problem is that we have a custom scope (in an Android activity) and in some fragments presenters we use this scope to retrieve dependencies. Here is how it looks like at the usage site (in a presenter):

```
val SCOPE_ID = "BookingActivity"
private val bookingPresenter: BookingPresenter
  get() = getKoin().getScope(SCOPE_ID).get()
```
The scope is created like this (in the activity):
```
scope = getKoin().createScope(SCOPE_ID, named<BookingActivity>())
```
I tried to mock it like the following, but `ScopeRegistry` throws a exception `No scope definition found for scopeName`:
```
startKoin {
  module {
    scope(named("BookingActivity")) {
      scoped { bookingPresenter }
    }
  }
}
getKoin().getOrCreateScope("BookingActivity", named<BookingActivity>())
```
I also tried without creating the scope explicitly in the test (in the last line) and tried using the `declare` function, but the result is the same. 

Is there a way to workaround this?

**Comment by pedrodimoura** [comment]
@Yazazzello @sadeqzadeh Try using `scope.declareMock<T>()` available on #546

**Comment by arnaudgiuliani** [comment]
`2.0.2` will be publish soonly.

**Comment by Yazazzello** [comment]
@pedrodimoura thanks!

**Comment by arnaudgiuliani** [comment]
available in 2.1.0-alpha-6 and after
